WalletWasabi.Gui/Tabs/LegalDocumentsViewModel.cs

```diff
@@ -1,15 +1,58 @@
 using System;
 using System.IO;
+using System.Reactive;
+using System.Reactive.Linq;
 using Avalonia;
+using Avalonia.Logging;
 using Avalonia.Platform;
+using AvalonStudio.Extensibility;
+using AvalonStudio.Shell;
+using ReactiveUI;
+using Splat;
+using WalletWasabi.Gui.Helpers;
+using WalletWasabi.Gui.Models.TextResourcing;
 using WalletWasabi.Gui.ViewModels;
+using WalletWasabi.Legal;
 
 namespace WalletWasabi.Gui.Tabs
 {
 	public class LegalDocumentsViewModel : TextResourceViewModelBase
 	{
-		public LegalDocumentsViewModel() : base("Legal Documents", new Uri("avares://WalletWasabi.Gui/Assets/LegalDocuments.txt"))
+		public ReactiveCommand<Unit, Unit> AgreeClicked { get; set; }
+		public LegalDocuments LegalDoc { get; }
+
+		public bool IsAgreed { get; set; }
+
+		public LegalDocumentsViewModel(string content = null, LegalDocuments legalDoc = null) : base("Legal Documents", new TextResource() { FilePath = legalDoc?.FilePath, Content = content })
+		{
+			LegalDoc = legalDoc;
+			IsAgreed = content is null; // If content wasn't provided, then the filepath must had been provided. If the file exists, then it's agreed.
+
+			AgreeClicked = ReactiveCommand.CreateFromTask(async () =>
+			{
+				IsAgreed = true;
+				await LegalDoc.ToFileAsync(TextResource.Content);
```

Comment:
Agreeing with end up a file being written. I would store the downloaded file somewhere and on an agreement, I would make a flag. With this you can separate the update procedure to steps:

1. Check the current legal docs. If it is not there copy it from Assets and use the version of that.
2. Updatechecker runs and if there is a new version download it to disagreed storage.
3. If it is a silent change legal doc, just replace the agreed file. end of the procedure. 
4. If it is a new docs, notify the user about the new version.
5. Pop up the agreement screen
6. Put the flag that the user agreed on the current version. 

Flag can be: 
- A file in a specific folder.
- An AgreedLegalDocsVersion="something" somewhere
- Any other solution.

Reply:
I thought we default configureawaiting in the library project and default not configureawaiting in the ui project.

Reply:
> Check the current legal docs. If it is not there copy it from Assets and use the version of that.

That is not good as described here: https://github.com/zkSNACKs/WalletWasabi/pull/2805#issuecomment-575685274

> Updatechecker runs and if there is a new version download it to disagreed storage.

Idk, if you want to implement it that should be fine, but it's messy.

> If it is a silent change legal doc, just replace the agreed file. end of the procedure.

We should not have the ability of "silently making the user agree into something." If it's a typo fix, it doesn't have to be on the user's computer, we can just fix it, if it's a change in the meaning, then we must increase the version and make the user agree not silently.

> If it is a new docs, notify the user about the new version.
> Pop up the agreement screen

It works that way.

> Put the flag that the user agreed on the current version.

Currently the flag is the existence of the file.